core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -177,7 +187,11 @@ public final T directExecutor() {
 
   @Override
   public final T executor(Executor executor) {
-    this.executor = executor;
+    if (executor != null) {
+      this.executorPool = new FixedObjectPool<Executor>(executor);
```

Comment:
SharedResourcePool is somewhat similar to a pool, but I agree it still isn't really. The Pool API and semantics do actually match a pool pretty well, though, so I'm having trouble coming up with an alternative name. I'm open for a rename in the future.